src/pj_obs_api.c

```diff
@@ -222,41 +266,150 @@ PJ_COORD proj_trans_coord (PJ *P, enum proj_direction direction, PJ_COORD coo) {
 }
 
 
-/* Measure numerical deviation after n roundtrips fwd-inv (or inv-fwd) */
-double proj_roundtrip (PJ *P, enum proj_direction direction, int n, PJ_OBS obs) {
-    int i;
-    PJ_OBS o, u;
 
+/*************************************************************************************/
+int proj_transform (
+    PJ *P,
+    enum proj_direction direction,
+    size_t stride,
+    double *x, size_t nx,
+    double *y, size_t ny,
+    double *z, size_t nz,
+    double *t, size_t nt
+) {
+/**************************************************************************************
+
+    Transform a series of coordinates, where the individual coordinate dimension
+    may be represented by an array that is either
+
+        1. fully populated
+        2. a null pointer and/or a length of zero, which will be treated as a
+           fully populated array of zeroes
+        3. of length one, i.e. a constant, which will be treated as a fully
+           populated array of that constant value
+
+    The stride represents the step length, in bytes, between consecutive elements
```

Comment:
Sounds like a good idea, although it will stretch an already long parameter list a bit more. I'm sure you have seen many more real world use cases than I have, so I will look into it.

Would this be useful for use cases in GDAL, or is it more a general comment? For example, I agree that in coming years, we probably will see quite a few cases of "grafting elevation support onto legacy code", by adding a separate elevation array to an established 2D code base :-)